pandas/core/groupby/generic.py

```diff
@@ -1759,7 +1761,39 @@ def _normalize_keyword_aggregation(kwargs):
         else:
             aggspec[column] = [aggfunc]
         order.append((column, com.get_callable_name(aggfunc) or aggfunc))
-    return aggspec, columns, order
+
+    # uniquify aggfunc name if duplicated in order list
+    uniquified_order = _uniquify_aggfunc(order)
+
+    # GH 25719, due to aggspec will change the order of assigned columns in aggregation
+    # uniquified_aggspec will store uniquified order list and will compare it with order
+    # based on index
+    aggspec_order = [
+        (column, com.get_callable_name(aggfunc) or aggfunc)
+        for column, aggfuncs in aggspec.items()
+        for aggfunc in aggfuncs
+    ]
+    uniquified_aggspec = _uniquify_aggfunc(aggspec_order)
+
+    # get the new indice of columns by comparison
+    col_idx_order = [uniquified_aggspec.index(o) for o in uniquified_order]
+    return aggspec, columns, col_idx_order
+
+
+def _uniquify_aggfunc(seq):
+    """Uniquify aggfunc name of the pairs in the order list
+
+    Examples:
+    --------
+    >>> _uniquify_aggfunc([('a', '<lambda>'), ('a', '<lambda>'), ('b', '<lambda>')])
+    [('a', '<lambda>_0'), ('a', '<lambda>_1'), ('b', '<lambda>')]
```

Comment:
Is this example correct? I thought the names were mangled like

`<lambda_0>` rather than `<lambda>_0`.

Reply:
yeah, this example is correct unfortunately 😅  i could also make it to `<lambda_0>`, since this is only for mangling and not used somewhere else, and i thought just adding it as suffix at the end of string `<lambda>` is the easiest way. Because the most important thing here is to make sure for each column, this `<lambda>` with number is correct regardless it's in the middle or tail. But if you have preference in changing it, I will change!

Reply:
emm, i have a second thought on it, one reason to do `<lambda>_0` alike is I am not sure how to do it without `regex`, and if just adding `_0` or `_1` etc in the end, we could also do it to `min_1` or `max_1` alike stuff if users are willing to use `agg(newcol1=('col', 'min'), newcol2=('col', 'min'))` and then solve https://github.com/pandas-dev/pandas/blob/ba94f9baaa1a802eee4820a1188db6501a231ae6/pandas/tests/groupby/aggregate/test_aggregate.py#L443 issue in another PR, any thoughts? @TomAugspurger

Reply:
If the behavior is correct for the end-user, then I'm in favor of whatever is simplest. I just assumed that this would have to match, but I haven't read the implementation closely yet.

Reply:
yeah, behavior is correct for end-user for sure, this implementation is just to get correct order. Feel free to take a look if you have time, and thanks again for quick reply! @TomAugspurger